samples/unit-tests/legend/legend-general/demo.js

```diff
@@ -130,6 +130,18 @@ QUnit.test('Legend redraws', function (assert) {
     chart.series[0].hide(true);
 
     assert.strictEqual(visible, false, 'Legend item text has changed (#2165)');
+
+    chart.series[0].remove();
+
+    chart.update({
```

Comment:
You redraw the chart twice here. Instead, you can fire `series.remove()` method with a `redraw` flag set to `false` to update only options, and the chart will be redrawn only once during the `chart.update()` method.
https://api.highcharts.com/class-reference/Highcharts.Series#remove (see a `redraw` argument)